packages/core/src/app/shipping/Shipping.tsx

```diff
@@ -232,7 +232,7 @@ class Shipping extends Component<ShippingProps & WithCheckoutShippingProps, Ship
         const promises: Array<Promise<CheckoutSelectors>> = [];
         const hasRemoteBilling = this.hasRemoteBilling(methodId);
 
-        if (!isEqualAddress(updatedShippingAddress, shippingAddress)) {
+        if (!isEqualAddress(updatedShippingAddress, shippingAddress) || shippingAddress?.shouldSaveAddress !== updatedShippingAddress?.shouldSaveAddress) {
```

Comment:
When we compare two addresses in `isEqualAddress`, we explicitly ignore `shouldSaveAddress`, so I put the check here.
https://github.com/bigcommerce/checkout-js/blob/5f25ba2e45ec698bddd842b639f8a8b0b6cd92fa/packages/core/src/app/address/isEqualAddress.ts#L45